### ISSUE TITLE
PHP 7.4: Trying to access array offset on value of type null

### ISSUE THREAD

**Issue by iotch** [composer/composer#8324]
Getting `Trying to access array offset on value of type null` error exception with composer 1.9 and php 7.4 revision rb7c353c on Windows 10. 
This happens on required packages with no svn configured, particularly, on packages without git with at least one commit.

Output of `composer diagnose`:

```
Checking composer.json: OK
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys:
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
Composer version: 1.9.0
PHP version: 7.4.0-dev
```

When I run this command:

```
composer install
```

I get the following output:

```
                                                       
  [ErrorException]                                     
  Trying to access array offset on value of type null  
                                                       

Exception trace:
 () at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Repository/PathRepository.php:158
 Composer\Util\ErrorHandler::handle() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Repository/PathRepository.php:158
 Composer\Repository\PathRepository->initialize() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Repository/ArrayRepository.php:185
 Composer\Repository\ArrayRepository->getPackages() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/DependencyResolver/Pool.php:104
 Composer\DependencyResolver\Pool->addRepository() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Installer.php:379
 Composer\Installer->doInstall() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Installer.php:229
 Composer\Installer->run() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Command/UpdateCommand.php:163
 Composer\Command\UpdateCommand->execute() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Command/Command.php:245
 Symfony\Component\Console\Command\Command->run() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Application.php:835
 Symfony\Component\Console\Application->doRunCommand() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Application.php:185
 Symfony\Component\Console\Application->doRun() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Console/Application.php:258
 Composer\Console\Application->doRun() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Application.php:117
 Symfony\Component\Console\Application->run() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Console/Application.php:104
 Composer\Console\Application->run() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/bin/composer:61
 require() at C:\ProgramData\ComposerSetup\bin\composer.phar:24

```

**Comment by stof** [comment]
please share what your composer.json looks like.

**Comment by iotch** [comment]
I'm sure it is not relevant, however it is very simple, like this:

```
{
    "name": "acme/core",
    "description": "acme core module",
    "type": "library",
    "license": "proprietary",
    "minimum-stability": "dev",
    "prefer-stable": true,
    "autoload": {
        "psr-4": {
            "Acme\\Core\\": "src"
        }
    },
    "require": {
        "php": ">=7.4",
        "ext-json": "*",
        "acme/utils": "1.0"
    }
}
```

The error disappears if "acme/utils" package has at least one commit.

**Comment by stof** [comment]
Well, this is weird. The stack trace locates the error in the PathRepository class, but your composer.json does not have a `path` repository at all.

**Comment by Seldaek** [comment]
Fixed by #8296 which isn't in a release yet. You can use `composer self-update --snapshot` for now to get the fix.

**Comment by waqasraza123** [comment]
composer self-update --stable solved my issue.
